### DISCUSSION TITLE
Redirect function from database list

### DISCUSSION THREAD

**Discussion by gulgil** [OrchardCMS/OrchardCore#13090] [Q&A] [answered]
Hi all. 

First of all, I'm new to OrchardCore so I don't have the full knowledge of it's "inner workings" :)

I'm trying to migrate my old site to OrchardCore, and it's going ok. My current problem is that I have a DB table with about 500 rows of redirects, and I need some way of handling them. It's just two colums (OldPath, NewPath). 
It's a list of old news, content, articles pages etc., that don't exists anymore. And we want them to point to existing pages Multiple old paths can point to the same newpath. 

It looks like Orchard Workflows handles redirects, but I have no idea how to get that to read my table.

I was thinking about making a module and hooking into Orchard routing routine. I found this git https://github.com/Hazzamanic/Hazza.Routes
But that looks like it's made for an older Orchard.

What would be my best course of action?

**Comment by Piedone** [comment, marked as answer]
If I understand correctly, you want to have HTTP redirects from some old URLs to new ones. This is not something that you'd need Workflows, or Orchard for. Instead, you can do that from an ASP.NET Core middleware, like [the built-in one with `AddRedirect()`](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/url-rewriting?view=aspnetcore-7.0#url-redirect).

**Reply by gulgil** [reply]
Thanks Piedone. 
Thought I needed to use Orchard functionality for this, but it looks like I can use .net :)

**Comment by emrahtokalak** [comment]
https://github.com/EtchUK/Etch.OrchardCore.SEO 

You can check this repo, it contains a good redirection solution.
